ols/app/endpoints/feedback.py

```diff
@@ -1,19 +1,111 @@
 
+import json
 import logging
+from pathlib import Path
 from typing import Any
 
 from fastapi import APIRouter, Depends
 
-from ols.app.models.models import FeedbackRequest, FeedbackResponse
+from ols.app.models.models import (
+    FeedbackRequest,
+    FeedbackResponse,
+    RemoveFeedbackRequest,
+)
+from ols.utils import config
 from ols.utils.auth_dependency import auth_dependency
 
 logger = logging.getLogger(__name__)
 router = APIRouter(prefix="/feedback", tags=["feedback"])
 
 
+def list_feedbacks() -> list[str]:
+    """List feedbacks in the local filesystem.
+
+    Returns:
+        List of feedback files (without the ".json" extension).
+    """
+    storage_path = Path(config.ols_config.feedback_storage_location)
+    feedback_files = list(storage_path.glob("*.json"))
+    # extensions are trimmed, eg. ["12345678-abcd-0000-0123-456789abcdef", ...]
+    feedbacks = [f.stem for f in feedback_files]
+    logger.info(f"'{len(feedbacks)}' feedbacks found")
+    return feedbacks
+
+
+def store_feedback(conversation_id: str, user_id: str, feedback: dict) -> None:
+    """Store feedback in the local filesystem.
+
+    Args:
+        conversation_id: The conversation ID (UUID).
+        user_id: The user ID (UUID).
+        feedback: The feedback object.
+    """
+    # ensures storage path exists
+    storage_path = Path(config.ols_config.feedback_storage_location)
+    if not storage_path.exists():
+        storage_path.mkdir(parents=True)
+
+    data_to_store = {
+        "metadata": {
+            "conversation_id": conversation_id,
+            "user_id": user_id,
+        },
+        "feedback": feedback,
+    }
+
+    # stores feedback in a file named after the conversation id
+    feedback_file_path = storage_path / f"{conversation_id}.json"
+    if feedback_file_path.exists():
+        logger.warning(
+            f"feedback file '{feedback_file_path}' already exists, overwriting"
+        )
+    with open(feedback_file_path, "w", encoding="utf-8") as feedback_file:
+        json.dump(data_to_store, feedback_file)
+
+    logger.info(f"feedback '{conversation_id}' stored in '{feedback_file_path}'")
+
+
+def remove_feedback(feedback_id: str) -> None:
+    """Remove feedback from the local filesystem.
+
+    Args:
+        feedback_id: The feedback ID (UUID).
+    """
+    storage_path = Path(config.ols_config.feedback_storage_location)
+    feedback_file = storage_path / f"{feedback_id}.json"
+    if feedback_file.exists():
+        feedback_file.unlink()
+        if not feedback_file.exists():
+            logger.info(f"feedback file '{feedback_file}' removed")
+        else:
+            logger.error(f"feedback file '{feedback_file}' failed to remove")
+    else:
+        logger.error(f"feedback file '{feedback_file}' not found")
+
+
+# TODO: OLS-136 implements the collection mechanism - revisit the need
+# of this endpoint
+@router.get("/list")
+def get_user_feedbacks(auth: Any = Depends(auth_dependency)) -> FeedbackResponse:
```

Comment:
yeah i would prefer we not have this until we decide we need it(if ever).

also if we did want it, the auth protection needs to be different from the auth requirement for sending queries to OLS.  We wouldn't want a regular OLS user to be able to retrieve all the feedback objects.  So that is something to keep in mind as you're adding new endpoints, that different endpoints have different permission requirements.